Adding completed uploadVideo changes

dcelik opened the pull request:
This PR is complete functionality change in how uploadVideo works.
The following functionality/files have been added:
* app.py now has 100MB limits on both body and buffer size. This provides a limit for how big of requests people can send us before streaming would have to occur. This will help protect us as well.
* Added multipart_streamer.py module taken from [tornadostreamform](http://pythonhosted.org/tornadostreamform/) and modified it to work with Python 2.7 instead of 3.5 . It should be attributed and licensed correctly.
* uploadVideo now uses the MultiPartStreamer class to write stream chunks as they come in, rather than loading it into memory later. It changes the max_body_size parameter of the request in the prepare() method and is currently set to a conservative 20GB. We can increase/decrease this as needed to work with our max supported file size. 
* uploadVideo no longer supports passing of an identifier in order to reupload a Video. This functionality has been removed in favor of a new project being created for each video uploaded.
* pm.py's create_project function has been modified to instead pass in a TemporaryFileStreamedPart instead of creating a dictionary containing the raw binary. The .move command called will move the part to the passed file location and should be nearly instant due to the way os.move() works.
* statusHelper's clean up method has been modified to make sure that any file inside of project_dir that does not have a cfg file does not prevent the clean up from hitting all files 

See our [SantosGUI PR](https://github.com/santosfamilyfoundation/SantosGUI/pull/72) for more the accompanying client side changes.

Commit:
Adding completed uploadVideo changes

dcelik committed:
Update multipart_streamer.py

youralien commented on app/app.py:
```diff
@@ -44,8 +44,10 @@
-define("port", default=8888, help="run on the given port", type=int)
 
+define("port", default=8888, help="run on the given port", type=int)
+define("max_body_size", default=100*1024*1024, help="max size of content body", type=int)
```
I'd prefer a variable up top that says

```python
# number of bytes in a MegaByte
MB = 1024 * 1024
```
and then use it throughout the file

youralien commented on app/traffic_cloud_utils/statusHelper.py:
```diff
@@ -70,9 +70,12 @@ def mark_all_failed():
+                    if v == Status.Flag.IN_PROGRESS:
+                        StatusHelper.set_status(identifier, k, Status.Flag.FAILURE)
+            else:
+                print "Error: Could not get CFG file, not clearing this project"
```
Instead of "not clearing this project" say "could not mark project status failure flags for project {}".format(identifier)

youralien reviewed (approved):
Aside from stylistic comments, this is awesome work.  Big functionality change :)

dcelik committed:
Update statusHelper.py

dcelik committed:
Update app.py

dcelik committed:
Removed .temp folder

Now that streaming videos places them into the linux temp directory, we do not need our own .temp folder

dcelik commented:
Made changes suggested by Ryan.
